src/ColumnSchema.php

```diff
@@ -56,7 +56,7 @@ public function phpTypecast(mixed $value): mixed
         }
 
         if ($this->getType() === SchemaInterface::TYPE_JSON) {
-            return json_decode((string) $value, true, 512, JSON_THROW_ON_ERROR);
+            return is_array($value) ? $value : json_decode((string) $value, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
```suggestion
            return is_string($value) ? json_decode($value, true, 512, JSON_THROW_ON_ERROR) : $value;
```

Looks better check if it is string